### PR TITLE
Replace the fork version with upstream murmur3

### PR DESCRIPTION
**Pull request by koushiro**
Signed-off-by: koushiro <koushiro.cqx@gmail.com>

###  What have you changed?

Replace the fork version with upstream `murmur3`

###  What is the type of the changes?

- Engineering

###  How is the PR tested?

- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

**Comment by siddontang** [comment]
PTAL @BusyJay

File: src/coprocessor/statistics/cmsketch.rs [hunk 1]
```diff
@@ -29,25 +29,23 @@ impl CmSketch {
...
-        let mut out: [u8; 16] = [0; 16];
-        murmur3_x64_128(&mut bytes, 0, &mut out);
-        (
-            LittleEndian::read_u64(&out[0..8]),
-            LittleEndian::read_u64(&out[8..16]),
-        )
+    fn hash(mut bytes: &[u8]) -> Result<(u64, u64)> {
+        let out =
+            murmur3::murmur3_x64_128(&mut bytes, 0).map_err(|err| Error::Other(err.to_string()))?;
+        Ok((u64::from_le(out as u64), u64::from_le((out >> 64) as u64)))
```
Comment (BusyJay) [inline src/coprocessor/statistics/cmsketch.rs]: I don't think `from_le` is necessary.

File: src/coprocessor/statistics/cmsketch.rs [hunk 2]
```diff
@@ -29,25 +29,23 @@ impl CmSketch {
...
-    fn hash(mut bytes: &[u8]) -> (u64, u64) {
-        let mut out: [u8; 16] = [0; 16];
-        murmur3_x64_128(&mut bytes, 0, &mut out);
-        (
-            LittleEndian::read_u64(&out[0..8]),
-            LittleEndian::read_u64(&out[8..16]),
-        )
+    fn hash(mut bytes: &[u8]) -> Result<(u64, u64)> {
+        let out =
+            murmur3::murmur3_x64_128(&mut bytes, 0).map_err(|err| Error::Other(err.to_string()))?;
```
Comment (BusyJay) [inline src/coprocessor/statistics/cmsketch.rs]: Unwrap is OK as there can't be any IO error using bytes.

Comment (koushiro) [reply src/coprocessor/statistics/cmsketch.rs]: okay

File: src/coprocessor/statistics/cmsketch.rs (hunk 1 above)
Comment (koushiro) [reply src/coprocessor/statistics/cmsketch.rs]: Yeah, you're right

File: src/coprocessor/statistics/cmsketch.rs [hunk 3]
```diff
@@ -30,12 +29,8 @@ impl CmSketch {
 
     // `hash` hashes the data into two u64 using murmur hash.
     fn hash(mut bytes: &[u8]) -> (u64, u64) {
-        let mut out: [u8; 16] = [0; 16];
-        murmur3_x64_128(&mut bytes, 0, &mut out);
-        (
-            LittleEndian::read_u64(&out[0..8]),
```
Comment (BusyJay) [inline src/coprocessor/statistics/cmsketch.rs]: The order may be incorrect due to different endianness.

Comment (koushiro) [reply src/coprocessor/statistics/cmsketch.rs]: PTAL :rofl:

Comment (BusyJay) [reply src/coprocessor/statistics/cmsketch.rs]: After checking the source of `murmur3_x64_128`, the correct way should be "(out as u64, (out >> 64) as u64)". I used to think it use `u128::from_le` to compute the value, turns out it uses shift and or to combine two u64 into a u128.

Comment (koushiro) [reply src/coprocessor/statistics/cmsketch.rs]: It was my mistake. After comparing the fork version and `v0.5.1` carefully, I think it should be right now.

**Comment by BusyJay** [comment]
@breeswish PTAL

**Pull request merged**